### PR TITLE
chore: upgrade Jest and TypeScript dependencies to latest version

### PR DESCRIPTION
**Pull request by jonkoops**
**What kind of change does this PR introduce?**
- [ ] Bug Fix
- [ ] Feature
- [ ] Refactoring
- [ ] Style
- [ ] Build
- [x] Chore
- [ ] Documentation
- [ ] CI

**Did you add tests for your changes?**
- [ ] Yes, my code is well tested
- [x] Not relevant

**If relevant, did you update the documentation?**
- [ ] Yes, I've updated the documentation
- [x] Not relevant

**Summary**
Upgrades Jest and TypeScript dependencies to their respective latest version, and updated the Rollup configuration to match the latest conventions.

**Does this PR introduce a breaking change?**
No.

**Other information**
None.

### CODE REVIEW

**Comment by jonkoops** [comment]
This PR depends on the work in #73, so I have marked it as a draft until that is merged and this this one is rebased.

File: jest.config.js
Comment (jonkoops) [inline jest.config.js:1]: Went ahead and replaced this with the default config as generated by `ts-jest`. It seems the defaults are sensible enough for the existing setup to just work.

**Comment by rolandjitsu** [comment]
Just merged https://github.com/react-dropzone/file-selector/pull/73.

**Review by rolandjitsu — APPROVED** [review]

**Comment by jonkoops** [comment]
@rolandjitsu did a rebase, this one should be good to go now.

File: package.json [hunk 1]
```diff
@@ -45,22 +45,22 @@
     "test": "jest --watch"
   },
   "dependencies": {
-    "tslib": "^2.4.0"
+    "tslib": "^2.7.0"
```
Comment (rolandjitsu) [inline package.json:48]: I'd count this as a `feat` because we're bumping a runtime dependency.

**Comment by rolandjitsu** [comment]
> @rolandjitsu did a rebase, this one should be good to go now.

Because of the `tslib` bump, the commit becomes a `feat`.

**Comment by jonkoops** [comment]
Sure, you can amend the commit message when merging :+1:

**Pull request merged**